src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -314,29 +329,41 @@ protected override void ApplyParseOptionsChanged(ProjectId projectId, ParseOptio
                 throw new ArgumentNullException(nameof(options));
             }
 
-            var parseOptionsService = CurrentSolution.GetProject(projectId).LanguageServices.GetService<IParseOptionsService>();
-            Contract.ThrowIfNull(parseOptionsService, nameof(parseOptionsService));
+            var storage = ProjectPropertyStorage.Get(TryGetDTEProject(projectId), DeferredState.ServiceProvider);
 
-            string newVersion = parseOptionsService.GetLanguageVersion(options);
+            switch (options)
+            {
+                case CSharpCompilationOptions csharpOptions:
+                    storage.SetProperty("AllowUnsafeBlocks", nameof(ProjectConfigurationProperties3.AllowUnsafeBlocks),
```

Comment:
Why not use `nameof` for the first argument as well?

Reply:
This is intentional. As you can see in the other method below (LangVersion vs LanguageVersion), these names are not guaranteed to be the same. The fact that they happen to be same for this property is more of a coincidence, which is why I would prefer not using nameof here.

Reply:
Or do you know of any other interface that these build property names correspond to? (where I would see LangVersion for example)